src/extremeweatherbench/evaluate.py

```diff
@@ -302,24 +318,36 @@ def compute_case_operator(
         "Datasets built for case %s.", case_operator.case_metadata.case_id_number
     )
     results = []
-    for variables, metric in itertools.product(
-        zip(
-            case_operator.forecast.variables,
-            case_operator.target.variables,
-        ),
-        case_operator.metric_list,
-    ):
-        results.append(
-            _evaluate_metric_and_return_df(
-                forecast_ds=aligned_forecast_ds,
-                target_ds=aligned_target_ds,
-                forecast_variable=variables[0],
-                target_variable=variables[1],
-                metric=metric,
-                case_operator=case_operator,
-                **kwargs,
+
+    # Loop through metrics; if variables aren't defined for the metric, use variables
+    # declared in the InputBase objects
+    for metric in case_operator.metric_list:
+        # Determine which variable pairs to evaluate for this metric
+        if metric.forecast_variable is not None and metric.target_variable is not None:
```

Comment:
nit: should add explicit grouping to each clause of the logical

```suggestion
        if (metric.forecast_variable is not None) and (metric.target_variable is not None):
```